### ISSUE TITLE
[3.7.0.beta3] Call to a member function copyValue() on null

### ISSUE THREAD

**Issue by joshua-martin** [craftcms/cms#9080]
### Description

Getting the below error–it appears to be happening when I change entry types on an entry that already has content.

```
Error: Call to a member function copyValue() on null in /Users/joshuamartin/Documents/Websites/southwark-music/vendor/craftcms/cms/src/base/Element.php:2260
Stack trace:
#0 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/craftcms/cms/src/services/Elements.php(811): craft\base\Element->mergeCanonicalChanges()
#1 [internal function]: craft\services\Elements->craft\services\{closure}(Object(craft\db\Connection))
#2 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/yiisoft/yii2/db/Connection.php(810): call_user_func(Object(Closure), Object(craft\db\Connection))
#3 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/craftcms/cms/src/services/Elements.php(834): yii\db\Connection->transaction(Object(Closure))
#4 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/craftcms/cms/src/services/Drafts.php(328): craft\services\Elements->mergeCanonicalChanges(Object(craft\elements\Entry))
#5 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/craftcms/cms/src/controllers/EntryRevisionsController.php(476): craft\services\Drafts->publishDraft(Object(craft\elements\Entry))
#6 [internal function]: craft\controllers\EntryRevisionsController->actionPublishDraft()
#7 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#8 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/yiisoft/yii2/base/Controller.php(181): yii\base\InlineAction->runWithParams(Array)
#9 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/craftcms/cms/src/web/Controller.php(190): yii\base\Controller->runAction('publish-draft', Array)
#10 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/yiisoft/yii2/base/Module.php(534): craft\web\Controller->runAction('publish-draft', Array)
#11 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/craftcms/cms/src/web/Application.php(271): yii\base\Module->runAction('entry-revisions...', Array)
#12 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/craftcms/cms/src/web/Application.php(574): craft\web\Application->runAction('entry-revisions...', Array)
#13 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/craftcms/cms/src/web/Application.php(250): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#14 /Users/joshuamartin/Documents/Websites/southwark-music/vendor/yiisoft/yii2/base/Application.php(392): craft\web\Application->handleRequest(Object(craft\web\Request))
#15 /Users/joshuamartin/Documents/Websites/southwark-music/public_html/index.php(22): yii\base\Application->run()
#16 {main}
```

https://user-images.githubusercontent.com/1694957/123431142-d8a41080-d5c0-11eb-904b-1154375e5677.mp4

**Comment by brandonkelly** [comment]
Thanks for reporting that! Just fixed for the next release.

**Comment by brandonkelly** [comment]
Annnnd just released 3.7.0-beta.4 with that.
